modules/fedimint-mint-client/src/lib.rs

```diff
@@ -2306,25 +2320,28 @@ pub fn represent_amount<K>(
     for tier in tiers.tiers() {
         let notes = current_denominations.get(*tier);
         let missing_notes = u64::from(denomination_sets).saturating_sub(notes as u64);
-        let possible_notes = remaining_amount / *tier;
+        let possible_notes = remaining_amount / (*tier + fee_consensus.fee(*tier));
 
         let add_notes = min(possible_notes, missing_notes);
         denominations.inc(*tier, add_notes as usize);
-        remaining_amount -= *tier * add_notes;
+        remaining_amount -= (*tier + fee_consensus.fee(*tier)) * add_notes;
```

Comment:
`represent_amount` can be used for inputs or outputs right now, both require opposite handling of fees (adding vs subtracting). 

Maybe the `fee_consensus` argument can be made optional?

Reply:
How is it used for inputs? I only see it used for outputs and in the tests.

Reply:
I got confused, I think in the beginning there was some overlap, but ever since we target specific denominations that's not possible anymore anyway. I should rename the fn.